src/history.rs

```diff
@@ -196,6 +196,18 @@ pub fn rewrite_commit(
         parents,
     )?;
 
+    if let Ok((sig, _)) = repo.extract_signature(&base.id(), None) {
+        // Re-create the object with the original signature (which of course does not match any
+        // more, but this is needed to guarantee perfect round-trips).
```

Comment:
FWIW there is a comment a few lines up that needs updating now -- or we could entirely remove that equality check.

Reply:
I think that equality check can be removed. It's value as an optimization is super low as it is an quite rare case.